src/AutoConstructor.Generator/AutoConstructorGenerator.cs

```diff
@@ -23,142 +25,128 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
             i.AddSource(Source.InjectAttributeFullName, SourceText.From(Source.InjectAttributeText, Encoding.UTF8));
         });
 
-        IncrementalValuesProvider<ClassDeclarationSyntax> classDeclarations = context.SyntaxProvider
-                .CreateSyntaxProvider(static (s, _) => IsSyntaxTargetForGeneration(s), static (ctx, _) => GetSemanticTargetForGeneration(ctx))
-                .Where(static m => m is not null)!;
-
-        IncrementalValueProvider<(Compilation compilation, ImmutableArray<ClassDeclarationSyntax> classes, AnalyzerConfigOptions options)> valueProvider =
-            context.CompilationProvider
-            .Combine(classDeclarations.Collect())
+        IncrementalValuesProvider<GeneratorExectutionResult?> valueProvider = context.SyntaxProvider
+            .ForAttributeWithMetadataName(
+                Source.AttributeFullName,
+                static (node, _) => IsSyntaxTargetForGeneration(node),
+                static (context, _) => (ClassDeclarationSyntax)context.TargetNode)
+            .Where(static m => m is not null)
             .Combine(context.AnalyzerConfigOptionsProvider.Select((c, _) => c.GlobalOptions))
-            .Select((c, _) => (compilation: c.Left.Left, classes: c.Left.Right, options: c.Right));
+            .Combine(context.CompilationProvider)
```

Comment:
That is unfortunately, still not incremental 😢 

cc @Sergio0694 as the original issue author :)

Reply:
Yeah this isn't correct either. @k94ll13nn3 you cannot capture any symbols/compilations into your incremental pipeline, as they're (1) not equatable, and (2) they will root compilations and keep them alive longer than necessary. You should build your entire incremental model directly from the FAWMN transform, and pass that through. Then you can combine _that_ with your analyzer options, and bring those two (fully incremental models) to the output node.

Reply:
I checked with a Datetime injected into my files, and it was only updated when I modified something that impacted the ouput, and when I modified something else the content was not regenerated. Is there anything more to check to be sure that it is really incremental ?

Reply:
@k94ll13nn3 It's likely that the generator was re-running everything up to the `Execute` method (i.e, the `Execute` method itself is executing with every change), but only the source production step wasn't run. That is why you see it not re-generating, but in fact a large part of the generator is still run for every change.

Reply:
I see, I was focused on not getting the `RegisterSourceOutput` to reexecute that I didn't think to check the previous steps

Reply:
Yeah, basically capturing symbols or compilations (or diagnostics) essentially breaks incremental pipelines up to the last step they're present (which may or may not be the output node, depending on what the pipeline looks like). In this case, your pipeline would've always been re-run up until that `Select` call. So steps downstream of that (ie. the output node) would be ok, but you'd still be repeating all that work every single time up until that point, unnecessarily.